server/sonar-process-monitor/src/main/java/org/sonar/application/SchedulerImpl.java

```diff
@@ -223,8 +222,16 @@ public void onAppStateOperational(ProcessId processId) {
 
   @Override
   public void onProcessState(ProcessId processId, Lifecycle.State to) {
-    if (to == Lifecycle.State.STOPPED) {
-      onProcessStop(processId);
+    switch (to) {
+      case STOPPED:
+        onProcessStop(processId);
+        break;
+      case STARTING:
+        stopCountDown.incrementAndGet();
```

Comment:
instead of waiting for listener event STARTING, I suggest to move this statement to tryToStartProcess():
```
try {
      process.start(() -> {
        JavaCommand command = commandSupplier.get();
        return javaProcessLauncher.launch(command);
      });
     stopCountDown.incrementAndGet();
    } catch (RuntimeException e) {
```